backend/src/v2/component/launcher_v2.go

```diff
@@ -856,6 +862,20 @@ func getPlaceholders(executorInput *pipelinespec.ExecutorInput) (placeholders ma
 		key := fmt.Sprintf(`{{$.inputs.artifacts['%s'].uri}}`, name)
 		placeholders[key] = inputArtifact.Uri
 
+		// If the artifact is marked as already in the workspace, map to the workspace path
+		// with the same shape as LocalPathForURI, but rebased under the workspace mount.
+		if inputArtifact.GetMetadata() != nil {
+			if v, ok := inputArtifact.GetMetadata().GetFields()["_kfp_workspace"]; ok && v.GetBoolValue() {
+				localPath, lerr := LocalWorkspacePathForURI(inputArtifact.Uri)
+				if lerr != nil {
+					return nil, fmt.Errorf("failed to get local workspace path for input artifact %q: %w", name, lerr)
+				}
+				key = fmt.Sprintf(`{{$.inputs.artifacts['%s'].path}}`, name)
+				placeholders[key] = localPath
```

Comment:
Could you update `needsWorkspaceMount` so that `if strings.HasPrefix(strVal.StringValue, component.WorkspaceMountPath) {` is `if strings.Contains(strVal.StringValue, component.WorkspaceMountPath) {`?

This is because the placeholder replacement logic is much less strict and we need `needsWorkspaceMount` to match.